### PR TITLE
CID1508860: double lock confusion

### LINKED ISSUE #10430: CID 1508860: Double lock
AIO.cc
```
529      if (op->thread == AIO_CALLBACK_THREAD_AIO) {
530        SCOPED_MUTEX_LOCK(lock, op->mutex, thr_info->mutex->thread_holding);
531        op->handleEvent(EVENT_NONE, nullptr);
   	12. Falling through to end of if statement.
   	24. Falling through to end of if statement.
   	37. Condition op->thread == NULL, taking true branch.
532      } else if (op->thread == AIO_CALLBACK_THREAD_ANY) {
533        eventProcessor.schedule_imm(op);
   	38. Falling through to end of if statement.
534      } else {
535        op->thread->schedule_imm(op);
536      }
   	25. lock: ink_mutex_acquire locks my_aio_req->aio_mutex. [[show details](https://scan6.scan.coverity.com/eventId=34764104-39&modelId=34764104-0&fileInstanceId=164784741&filePath=%2Finclude%2Ftscore%2Fink_mutex.h&fileStart=46&fileEnd=52)]
   	
CID 1508860 (#1-2 of 2): Double lock (LOCK)
39. double_lock: ink_mutex_acquire locks my_aio_req->aio_mutex while it is locked.
537      ink_mutex_acquire(&my_aio_req->aio_mutex);
```

### PR DESCRIPTION
**Pull request by cmcfarlen**
I wonder if coverity is confused by following the `current_req` pointer for the unlock call vs `my_aio_req` for the lock.  I removed the `current_req` variable as it isn't necessary.

I also moved the release call before incrementing metrics since the lock doesn't need to be held for that.

Fixes #10430

### CODE REVIEW

File: iocore/aio/AIO.cc [hunk 1]
```diff
@@ -442,8 +440,10 @@ AIOThreadInfo::aio_thread_main(AIOThreadInfo *thr_info)
 #ifdef AIO_STATS
```
Comment (ywkaras) [inline iocore/aio/AIO.cc:440]: How the outer `for` loop not an infinite loop?  The `break` on line 438 breaks out of the `do` loop only, no?

File: iocore/aio/AIO.cc [hunk 2]
```diff
@@ -442,8 +440,10 @@ AIOThreadInfo::aio_thread_main(AIOThreadInfo *thr_info)
...
       num_requests--;
       current_req->queued--;
-      ink_atomic_increment((int *)&current_req->pending, 1);
+      ink_atomic_increment((int *)&my_aio_req->pending, 1);
```
Comment (ywkaras) [inline iocore/aio/AIO.cc:443]: You could get rid of the C-style cast.

File: iocore/aio/AIO.cc [hunk 3]
```diff
@@ -423,17 +423,15 @@ ink_aio_thread_num_set(int thread_num)
 void *
 AIOThreadInfo::aio_thread_main(AIOThreadInfo *thr_info)
```
Comment (ywkaras) [inline iocore/aio/AIO.cc:424]: Not directly relevant to this PR, but `thr_info` is redundant, it's  the same as the `this` pointer in this function.

**Comment by ywkaras** [comment]
The AIO code seems scary bad in general.

File: iocore/aio/AIO.cc [hunk 4]
```diff
@@ -452,9 +452,8 @@ AIOThreadInfo::aio_thread_main(AIOThreadInfo *thr_info)
         Metrics::increment(aio_rsb.read_count);
         Metrics::increment(aio_rsb.kb_read, op->aiocb.aio_nbytes >> 10);
       }
-      ink_mutex_release(&current_req->aio_mutex);
       cache_op((AIOCallbackInternal *)op);
-      ink_atomic_increment(&current_req->requests_queued, -1);
+      ink_atomic_increment(&my_aio_req->requests_queued, -1);
 #ifdef AIO_STATS
       ink_atomic_increment((int *)&current_req->pending, -1);
```
Comment (ywkaras) [inline iocore/aio/AIO.cc:458]: It lives.

File: iocore/aio/AIO.cc [hunk 5]
```diff
@@ -442,8 +440,10 @@ AIOThreadInfo::aio_thread_main(AIOThreadInfo *thr_info)
 #ifdef AIO_STATS
       num_requests--;
       current_req->queued--;
-      ink_atomic_increment((int *)&current_req->pending, 1);
+      ink_atomic_increment((int *)&my_aio_req->pending, 1);
 #endif
+      ink_mutex_release(&my_aio_req->aio_mutex);
```
Comment (ywkaras) [inline iocore/aio/AIO.cc:445]: Releasing early looks OK.

File: iocore/aio/AIO.cc (hunk 4 above)
Comment (cmcfarlen) [reply iocore/aio/AIO.cc]: Yep, I missed some inside the conditional code.  Thanks!

**Review by ywkaras — APPROVED** [review]

**Pull request merged**